JenkinsTransport/EnumUtils.cs

```diff
@@ -62,7 +62,7 @@ public class EnumUtils
         /// <param name="color">the color of the build</param>
         public static IntegrationStatus GetIntegrationStatus(string color)
         {
-            return BuildStatusMap.ContainsKey(color) ? BuildStatusMap[color] : IntegrationStatus.Unknown;
+            return (color != null && BuildStatusMap.ContainsKey(color)) ? BuildStatusMap[color] : IntegrationStatus.Unknown;
```

Comment:
I'd prefer the null check here as a guard clause rather than increasing the size of this ternary. 

if (color == null)
    return IntegrationStatus.Unknown

Same in the other places